File: dotnet/Lexy.Poc/Lexy.Poc.Core/Parser/Tokens/TableSeparatorToken.cs
namespace Lexy.Poc.Core.Parser.Tokens
{
    public class TableSeparatorToken : ParsableToken
    {
        public TableSeparatorToken(TokenCharacter character) : base(character)
        {
        }

        public override ParseTokenResult Parse(TokenCharacter character, IParserContext context)
        {
            return ParseTokenResult.Finished(true);
        }

        public override ParseTokenResult Finalize(IParserContext parserContext)
        {
            return ParseTokenResult.Finished(true);
        }
    }
}

File: typescipt/src/parser/tokens/tableSeparatorToken.ts
import {ParsableToken} from "./parsableToken";
import {TokenCharacter} from "./tokenCharacter";
import {newParseTokenFinishedResult, ParseTokenResult} from "./parseTokenResult";
import {TokenType} from "./tokenType";

export function instanceOfTableSeparatorToken(object: any): object is TableSeparatorToken {
  return object?.tokenType == TokenType.TableSeparatorToken;
}

export function asTableSeparatorToken(object: any): TableSeparatorToken | null {
  return instanceOfTableSeparatorToken(object) ? object as TableSeparatorToken : null;
}

export class TableSeparatorToken extends ParsableToken {

  public tokenIsLiteral: boolean = false;
  public tokenType = TokenType.TableSeparatorToken;

  constructor(character: TokenCharacter) {
    super(character);
  }

  public parse(character: TokenCharacter): ParseTokenResult {
    return newParseTokenFinishedResult(true);
  }

  public finalize(): ParseTokenResult {
    return newParseTokenFinishedResult(true);
  }
}